src/node_url.cc

```diff
@@ -1653,9 +1653,10 @@ void URL::Parse(const char* input,
               state = kFragment;
               break;
             default:
-              if ((!IsWindowsDriveLetter(ch, p[1]) ||
-                   end - p == 1 ||
-                   (p[2] != '/' &&
+              if ((end - p == 1 ||
```

Comment:
Seems like they agree with me.

We can also calculate per iteration an actual *remaining* variable, like we are [already doing](https://github.com/nodejs/node/blob/4677766d210c2dfb19f9c8d930bba99b390fde1c/src/node_url.cc#L765) for some other functions in the file. This should eliminate any confusion around this variable. The most correct implementation would be something like `const size_t remaining = end == pointer ? 0 : (end - pointer - 1);`.